src/mailadm/bot.py

```diff
@@ -0,0 +1,92 @@
+import sys
+
+import deltachat
+from deltachat import account_hookimpl
+from mailadm.db import DB
+from mailadm.commands import add_user, add_token, list_tokens
+import os
+from threading import Event
+
+
+class SetupPlugin:
+    def __init__(self, admingrpid):
+        self.member_added = Event()
+        self.admingrpid = admingrpid
+        self.message_sent = Event()
+
+    @account_hookimpl
+    def ac_member_added(self, chat: deltachat.Chat, contact, actor, message):
+        assert chat.num_contacts() == 2
+        if chat.id == self.admingrpid:
+            self.member_added.set()
+
+    @account_hookimpl
+    def ac_message_delivered(self, message: deltachat.Message):
+        if not message.is_system_message():
+            self.message_sent.set()
+
+
+class AdmBot:
+    def __init__(self, db):
+        self.db = db
+        with self.db.read_connection() as conn:
+            config = conn.config
+            self.admingrpid = config.admingrpid
+
+    @account_hookimpl
+    def ac_incoming_message(self, message: deltachat.Message):
+        print("process_incoming message:", message)
```

Comment:
Somehow sending a message to the admin group doesn't trigger this hook. I don't understand why.

Log output:

```
7.37 [events-bot] DC_EVENT_INFO data1=0 data2=src/receive_imf.rs:98: Receiving message, seen=true...
7.39 [events-bot] DC_EVENT_INFO data1=0 data2=src/e2ee.rs:215: Detected Autocrypt-mime message
7.40 [events-bot] DC_EVENT_WARNING data1=0 data2=src/mimeparser.rs:732: Ignoring nested protected headers
7.40 [events-bot] DC_EVENT_INFO data1=0 data2=src/receive_imf.rs:120: received message has Message-Id: Gr.GXDWDX6Soc0.OUq6MtEnUho@systemli.org
7.40 [events-bot] DC_EVENT_INFO data1=0 data2=src/receive_imf.rs:1162: Message has 1 parts and is assigned to chat #Chat#12.
7.40 [events-bot] DC_EVENT_MSGS_CHANGED data1=12 data2=28
7.40 [events-bot] DC_EVENT_INFO data1=0 data2=src/imap.rs:904: 1 mails read from "INBOX".
7.45 [events-bot] DC_EVENT_CONNECTIVITY_CHANGED data1=0 data2=0
7.45 [events-bot] DC_EVENT_INFO data1=0 data2=src/imap.rs:1710: ignoring unsolicited response Recent(1)
7.53 [events-bot] DC_EVENT_INFO data1=0 data2=src/imap/idle.rs:57: INBOX: Idle entering wait-on-remote state
``` 

Any ideas?

Reply:
Maybe this will work when the admbot.sqlite deltachat database is persistent one day.

Reply:
I resolved it by using a riseup.net account for testing - the error seems to appear only if the bot account is on a mailcow server. :shrug: